recipe_scrapers/sunset.py

```diff
@@ -1,7 +1,39 @@
 from ._abstract import AbstractScraper
+from ._grouping_utils import IngredientGroup
 
 
 class Sunset(AbstractScraper):
     @classmethod
     def host(cls):
         return "sunset.com"
+
+    def ingredient_groups(self):
+        groups = []
+        current_group = []
+        current_purpose = None
+
+        for element in self.soup.select(
+            ".recipe-ingredients .cooked-single-ingredient"
+        ):
+            classes = element.get("class", [])
+            if "cooked-heading" in classes:
+                if current_group:
+                    groups.append(
+                        IngredientGroup(
+                            ingredients=current_group, purpose=current_purpose
+                        )
+                    )
+                current_group = []
+                current_purpose = element.get_text(strip=True)
+            elif "cooked-ingredient" in classes:
+                ingredient = " ".join(
+                    element.get_text(separator=" ", strip=True).split()
+                )
```

Comment:
Maybe nitpicky, but I think we could/should re-use `normalize_string` here:

```suggestion
                ingredient = normalize_string(element.get_text())
                )
```

(and maybe also for the heading names, for consistency)